superset/security/manager.py

```diff
@@ -409,15 +409,16 @@ def can_access_all_datasources(self) -> bool:
         :returns: Whether the user can access all the datasources
         """
 
-        return self.can_access("all_datasource_access", "all_datasource_access")
+        return self.can_access_all_databases() or self.can_access(
+            "all_datasource_access", "all_datasource_access"
+        )
```

Comment:
This makes sense and looks fine, I'm just worried about unintended consequences. Would be nice to get a signoff from @dpgaspar and/or @villebro and/or @john-bodley here, just in case.

Also, it would be really nice if we could decouple the data permissions from the API permissions. Maybe we should create a workgroup to brainstorm how this should look like.

Reply:
I agree, grepping for `all_database_access` there's very little justifying its existence as AFAIU, two choices here:
- make it work as "expected" based on name which would be what's in this PR, essentially making `all_database_access` == `all_datasource_access`
- deprecate `all_database_access` and [maybe] "migrate"/reassign to `all_datasource_access` where it's defined

Reply:
Digging a bit deeper I found a reference to `SecurityManager.can_access_all_databases` in `superset/databases/filters.py`, which seems potentially at odds with the model-based perms as in `Database.can_write` and `Database.can_read`. If we were to deprecate `all_database_access` we could transfer it to `Database.can_read` in roles where it's specified to try to match the current behavior.

Reply:
@eschutho mentioned the case where we want to give someone access to all the tables for them to query in SQL Lab, but we don't want them modifying the database configuration. In that case, we'd want them to have `all_datasource_access` but not `all_database_access`, no?

Reply:
I think the CRUD on Database is tied to the model / modelview, as opposed to this special term.

I have a spreadsheet of all the perms, and the relevant ones for database CRUD should be these ->

<google-sheets-html-origin><style type="text/css"><!--td {border: 1px solid #cccccc;}br {mso-data-placement:same-cell;}--></style>
Database | can_export | MODEL_VIEW
-- | -- | --
Database | can_read | MODEL_VIEW
Database | can_write | MODEL_VIEW
Databases | menu_access | MENU

Reply:
We (Airbnb) tend not to use these access controls and thus I don't really have much to add. My only question would be, why this hasn't been an issue in the past as it _potentially_ looks like a fairly egregious oversight from a consistency standpoint—if in fact the proposed behavior is correct.